### ISSUE TITLE
There is issues with send command api to the manage Device

### ISSUE THREAD

**Issue by vishalvk78** [micromdm/micromdm#923]
### What version of `micromdm` are you using?
MicroMDM 1.11

### What `micromdm` command did you run?
Below API:
r.Methods("POST").Path("/v1/commands").Handler(httptransport.NewServer(
		e.NewCommandEndpoint,
		decodeNewCommandRequest,
		httputil.EncodeJSONResponse,
		options...,
	))

### What did you expect to see?
It checking Basic Auth type athentication after sending response

### What did you see instead?
I want need to check Basic Auth type athentication before sending the response.

**Comment by korylprince** [comment]
The server checks authentication before the request is even decoded.

The basic auth middleware is [added to every API endpoint](https://github.com/micromdm/micromdm/blob/c7e70b94d0cfc7710e5c92be20d4534d9d5a0640/cmd/micromdm/serve.go#L276-L310).

[The middleware checks the auth before running the downstream handler.](https://github.com/go-kit/kit/blob/ca4112baa34cb55091301bdc13b1420a122b1b9e/auth/basic/middleware.go#L73-L91)

**Comment by vishalvk78** [comment]
yes, you are right but for this api this athentication is there in code, but this is not working properly for me.

If I hit the below api and giving wrong password in athentication and also wrong in input that i am passing throgh json format, normally that need to give unathorized error because need to check athhorization first then inputs are wrong or right ? try to hit this api once


r.Methods("POST").Path("/v1/commands").Handler(httptransport.NewServer(
		e.NewCommandEndpoint,
		decodeNewCommandRequest,
		httputil.EncodeJSONResponse,
		opt

**Comment by korylprince** [comment]
@vishalvk78 you're correct; I was mistaken in my first post. go-kit (the toolkit MicroMDM uses)'s version of middleware actually runs middlewares after decoding the request.

Which means if the request you send can't be properly decoded, it will fail at that step before failing authentication. The decoded requests are only sent to the handler if the middleware passes, so only the decoding is done before checking authentication.

**Comment by vishalvk78** [comment]
yes, right, but with the logic If I hit another api then it work properly like first checking authentication then decoding means if authentication is wrong as well as request data is wrong then it will give unathorized error.
for example, below api from apns,

	r.Methods("POST").Path("/v1/push/{udid}").Handler(httptransport.NewServer(
		e.PushEndpoint,
		decodePushRequest,
		httputil.EncodeJSONResponse,
		options...,
	))
	
I think this is bug with the command api that i am facing.

**Comment by korylprince** [comment]
All of the API endpoints follow the same basic path:

* decode request
* authentication middleware
* request processing (using decoded request)

For the majority of the endpoints, the initial decoding doesn't have any additional logic beyond parsing the JSON. The command endpoint _does_ have a lot of extra logic to verify the command is valid. So while it may appear the command endpoint is different than the rest, it's technically doing things in the same order.

I agree this is not ideal. However, I would not expect this to change in MicroMDM v1.
